kura/org.eclipse.kura.linux.command/src/main/java/org/eclipse/kura/linux/command/CommandServiceImpl.java

```diff
@@ -8,170 +8,82 @@
  *
  * Contributors:
  *     Eurotech
+ *     Red Hat Inc
  *******************************************************************************/
 package org.eclipse.kura.linux.command;
 
-import java.io.BufferedReader;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.attribute.PosixFilePermission.OWNER_EXECUTE;
+import static java.nio.file.attribute.PosixFilePermission.OWNER_READ;
+import static java.nio.file.attribute.PosixFilePermission.OWNER_WRITE;
+
+import java.io.ByteArrayOutputStream;
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.OutputStream;
 import java.io.PrintWriter;
-
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.EnumSet;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteException;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.eclipse.kura.KuraErrorCode;
 import org.eclipse.kura.KuraException;
 import org.eclipse.kura.command.CommandService;
-import org.eclipse.kura.core.util.ProcessUtil;
-import org.eclipse.kura.core.util.SafeProcess;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class CommandServiceImpl implements CommandService {
 
-    private static final Logger s_logger = LoggerFactory.getLogger(CommandServiceImpl.class);
-
-    private static final String SCRIPT_FILE = System.getProperty("java.io.tmpdir") + File.separator + "runCmd.sh";
-
-    private File m_scriptFile;
-
-    // ----------------------------------------------------------------
-    //
-    // Activation APIs
-    //
-    // ----------------------------------------------------------------
-    protected void activate() {
-        s_logger.debug("Activating...");
-    }
-
-    protected void deactivate() {
-        s_logger.debug("Deactivating...");
-    }
-
-    // ----------------------------------------------------------------
-    //
-    // Service APIs
-    //
-    // ----------------------------------------------------------------
+    private static final File TEMP_DIR = new File(System.getProperty("java.io.tmpdir"));
 
     @Override
-    public String execute(String cmd) throws KuraException {
-        if (cmd == null) {
-            s_logger.debug("null command");
-            return "null command";
+    public String execute(final String cmd) throws KuraException {
+        
+        if (cmd == null || cmd.isEmpty()) {
+            return "<empty command>";
         }
-
-        // Delete script file if it exists
-        this.m_scriptFile = new File(SCRIPT_FILE);
-        if (this.m_scriptFile.exists()) {
+        
+        try {
```

Comment:
The finally section can throw an IOException as well, which is this way covered by the outer try-catch